### PR TITLE
Enable MQTT5 client unit tests in CI and improve MQTT5 client error handling

### PR DESCRIPTION
**Pull request by xiazhvera**
*Issue #, if available:*

*Description of changes:*
1. As the latest c-common change added a way to skip a test. We enable the Mqtt5 Unit tests in the CI and skip it if the environment variable is not set
2. The Mqtt5 client and builder will now return nullptr on failure. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/Mqtt5ClientTest.cpp [hunk 1]
```diff
@@ -101,11 +101,17 @@ struct Mqtt5TestEnvVars
             {
                 m_error |= aws_get_environment_value(allocator, s_mqtt5_test_envName_direct_hostName, &m_hostname);
                 m_error |= aws_get_environment_value(allocator, s_mqtt5_test_envName_direct_port, &m_port);
-                if (m_error == AWS_OP_SUCCESS)
+                if (m_error != AWS_OP_SUCCESS)
                 {
-                    m_hostname_string = aws_string_c_str(m_hostname);
-                    m_port_value = static_cast<uint16_t>(atoi(aws_string_c_str(m_port)));
+                    return;
```
Comment (TwistedTwigleg) [inline tests/Mqtt5ClientTest.cpp:106]: Should these empty `return` statements return `AWS_OP_SKIP`?

File: tests/Mqtt5ClientTest.cpp [hunk 2]
```diff
@@ -101,11 +101,17 @@ struct Mqtt5TestEnvVars
...
+                }
+                if (m_hostname == NULL || m_port == NULL)
+                {
+                    m_error = AWS_OP_ERR;
+                    return;
```
Comment (TwistedTwigleg) [inline tests/Mqtt5ClientTest.cpp:111]: Return `m_error`?

**Review by TwistedTwigleg — APPROVED** [review]

File: tests/Mqtt5ClientTest.cpp (hunk 1 above)
Comment (xiazhvera) [reply tests/Mqtt5ClientTest.cpp:106]: The environment variables are handled in the constructor of `Mqtt5TestEnvVars`, there is nothing to return. The `Mqtt5TestEnvVars` is used to handle the environment vars. We will check the m_error to determine if it successfully process the ENV.

File: tests/Mqtt5ClientTest.cpp (hunk 2 above)
Comment (xiazhvera) [reply tests/Mqtt5ClientTest.cpp:111]: Same to the previous case.

**Pull request merged**